packages/workspace/src/browser/workspace-service.ts

```diff
@@ -401,3 +467,53 @@ export interface WorkspaceInput {
     preserveWindow?: boolean;
 
 }
+
+interface WorkspaceData {
+    folders: Array<{ path: string }>;
+    // TODO add workspace settings settings?: { [id: string]: any };
+}
+
+namespace WorkspaceData {
+    // tslint:disable-next-line:no-any
+    export function is(data: any): data is WorkspaceData {
+        return data && data.folders !== undefined && Array.isArray(data.folders);
```

Comment:
we should really validate against the schema with ajx, as for preferences and keymaps